File: Segundo Trimestre/2- Segundos Ejercicios POO/Java/terminales/Terminal.java
/**
 * Implementa la clase Terminal. Un terminal tiene asociado un número. Los 
 * terminales se pueden llamar unos a otros y el tiempo de conversación corre
 * para ambos. A continuación se proporciona el contenido del main y el resultado
 * que debe aparecer por pantalla.
 * 
 * @author Rafael López | RafaLpeC
 */

package terminales;

public class Terminal {
	private String numero;
	/**
	 * Número de segundos acumulados de cada una de las llamadas entre terminales
	 */
	private double segundos;
	/**
	 * @param numero
	 */
	public Terminal(String numero) {
		setNumero(numero);
		setSegundos(0);
	}
	
	/**
	 * @return the numero
	 */
	String getNumero() {
		return numero;
	}
	/**
	 * @param numero the numero to set
	 */
	private void setNumero(String numero) {
		this.numero = numero;
	}
	/**
	 * @return the segundos
	 */
	double getSegundos() {
		return segundos;
	}
	/**
	 * Incrementa los segundos de cada terminal
	 * @param segundos the segundos to set
	 */
	private void setSegundos(double segundos) {
		this.segundos += segundos;
	}
	
	public void llama(Terminal t2, double segundos) {
		this.setSegundos(segundos);

	}

	/* (non-Javadoc)
	 * @see java.lang.Object#toString()
	 */
	@Override
	public String toString() {
		return "Nº " + getNumero() + " - " + getSegundos() + "s de conversación";
	}
}

File: Segundo Trimestre/2- Segundos Ejercicios POO/Python/terminales/Terminal.py
'''
Implementa la clase Terminal. Un terminal tiene asociado un número. Los 
terminales se pueden llamar unos a otros y el tiempo de conversación corre
para ambos. A continuación se proporciona el contenido del main y el resultado
que debe aparecer por pantalla.

@author: Rafael López | RafaLpeC
'''

class Terminal():
    
    def __init__(self,numero):
        self.__numero = numero;
        self.__segundos=0;
    
    def getNumero(self):
        return self.__numero;
    
    def __setNumero(self,numero):
        self.__numero= numero;
    
    def getSegundos(self):
        return self.__segundos;
    
    def setSegundos(self,segundos):
        self.__segundos+=segundos;
    
    def llama(self, t2, segundos):
        self.setSegundos(segundos);
        t2.setSegundos(segundos);
        
    def __str__(self):
        return "Nº " + str(self.getNumero()) + " - " + str(self.getSegundos()) + "s de conversación"
    